### ISSUE TITLE
Search results: Paging doesn't work when using a page query string parameter

### ISSUE THREAD

**Issue by ileana2019** [microsoft-search/pnp-modern-search#2278]
**Version used**
Ex: 4.7.0

**Describe the bug**
On a SharePoint home page, I have a link to the search results page. The link contains a query parameter: ex: https://tata.sharepoint.com/sites/coco/SitePages/Search%20Results.aspx?Customer=blabla. On the search results page I have the search results webpart, that is configured to show paging, and to filter the results based on the Customer query string parameter:
![Search results properties](https://user-images.githubusercontent.com/55545989/183929538-ded7da2d-42ec-4955-9cec-87f5a0518fc9.PNG)

When clicking on page 2, at the bottom of the search results, nothing happens.

**To Reproduce**
1. Create a page that contains the search results webpart
2. Add a query string parameter to the page url (ex: Customer=blabla)
3. Configure the search results webpart to filter the results based on the query string parameter value (see screenshot above)
4. Save the webpart configuration and click on page 2 of the search results
5. You still see page 1 of the search results

**Expected behavior**
Display page 2 of the search results.

**Desktop (please complete the following information):**
 - Browser: Chrome

**Comment by FranckyC** [comment]
@ileana2019 your configuration seems wrong. To use the query string parameter, you must use the "Use input query text" option in the connection page of the target Search Results WP like this:

![image](https://user-images.githubusercontent.com/5659122/184045507-47d146fa-7080-4dbe-9063-f9687474fb91.png)

The option "Connect to a search results Web Part" is only relevant when you connect two Search Results WPs together via item selection.

**Comment by ileana2019** [comment]
HI @FranckyC ,
On the search results page, I also have a search box webpart, which I connect to the search results webpart with this settings:
![Search results](https://user-images.githubusercontent.com/55545989/184117044-4bc6b669-4a43-40ae-96fd-7138a9c7df88.PNG)
On that page I also have the filters and verticals webparts; so the entire package. The configuration that I did on the webpart was the only configuration that worked to have everything (query string parameter, search box, filters, verticals) connected to the search results webpart. 
Still, the issue remains that the query string parameter and paging do not work together.

**Comment by FranckyC** [comment]
@ileana2019 I'm not sure what you are trying to achieve here. If you use a query string parameter to query content dynamically at page load, the correct setup is the one showed in my previous screenshot. This is different from filters (aka refiners). If you want filters to be able at page load, then you must use [filters deep link](https://microsoft-search.github.io/pnp-modern-search/usage/search-filters/#filter-deep-linking) instead. The screenshot of the configuration you posted is not valid (your 'step 3') for your use case.

**Comment by ferrarirosso** [comment]
We encountered the same issue :
1. Create a page where you reference your search result page with deeplinks and some filters
2. Navigate to this page, if you have more than one page of results, the first "paging" will not work (you stay on the same page), subsequent paging works.

The behaviour is caused by https://github.com/microsoft-search/pnp-modern-search/blob/main/search-parts/src/webparts/searchResults/SearchResultsWebPart.ts, Method getDataContext, selectedFilters and _lastSelectedFilters are compared, if there is a mismatch, the code set the page number to 1.
I have attached a screenshot of these fields :
![Screenshot 2022-11-01 at 18 23 14](https://user-images.githubusercontent.com/5904018/199333426-a61d4ffe-253f-4f65-b03c-93d496f1b9c0.png)
The count & disable value are missing
I added a PR to address this problem https://github.com/microsoft-search/pnp-modern-search/pull/2503
